arbnode/batch_poster.go

```diff
@@ -502,8 +502,7 @@ func (b *BatchPoster) addEspressoBlockMerkleProof(
 			return err
 		}
 
-		if jst.Header.Height == 0 {
-			// This means the header in the jst is still the dummy header.
+		if jst.Header == nil {
```

Comment:
@ImJeremyHe what are the implications / reasons for making the header optional?

Reply:
Just want to get rid of using the dummy header. In the previous version, we fetch transactions from hotshot blocks, so hotshot header is always there. So this is not optional.

In sovereign sequencer, the sequencer builds a dummy jst as a placeholder for easily decoding later.

(The reason for the dummy jst is largely because we can reuse the code here, facilitating the tx streamer to tell the reason of different message bytes. Are we really reorging, or just updating the jst.)
https://github.com/EspressoSystems/nitro-espresso-integration/blob/7091d0732e748ea4e3b230a2f0b19985b7a9d34c/arbnode/transaction_streamer.go#L791-L793

We currently are using a dummy header to build the dummy jst.
The data of the dummy header is a noise actually and it takes time to decode and encode it, though it is not a big issue.